### DISCUSSION TITLE
Joomla 4 cli core:install doesn't work  in Docker

### DISCUSSION THREAD

**Discussion by nickyeoman** [joomla/joomla-cms#43434] [Q&A]
### Steps to reproduce the issue

1. Create a Joomla container in Docker using the image `joomla:4.3.1` from [Docker Hub](https://hub.docker.com/_/joomla).
2. Try to run the command `php cli/joomla.php core:install` inside the container, following the instructions outlined in the [Joomla CLI Update documentation](https://docs.joomla.org/J4.x:CLI_Update#Installing_Joomla.21_via_the_CLI).

### Expected result

The Joomla installer should run successfully.

### Actual result

The output displays: `Install Joomla to run cli commands`. This behavior is expected according to the code found in [cli/joomla.php](https://github.com/joomla/joomla-cms/blob/4.3-dev/cli/joomla.php) on line 46:

```php
// Check if installed
if (
    !file_exists(JPATH_CONFIGURATION . '/configuration.php')
    || (filesize(JPATH_CONFIGURATION . '/configuration.php') < 10)
) {
    echo 'Install Joomla to run cli commands' . PHP_EOL;
    exit;
}
```

### System information

- Docker image: `joomla:4.3.1` from [Docker Hub](https://hub.docker.com/_/joomla)

### Additional comments

According to the [Installing Joomla via the CLI](https://docs.joomla.org/J4.x:CLI_Update#Installing_Joomla.21_via_the_CLI) documentation, it should be possible to install Joomla using Joomla cli. However, this is currently not working because the `cli/joomla.php` file checks for the existence of `configuration.php` before attempting the installation. This check prevents the installation process when trying to create the `configuration.php` file in the first place.

**Comment by alikon** [comment]
please follow this https://docs.joomla.org/J4.x:Joomla_CLI_Installation

**Comment by Fedik** [comment]
Hi,

> According to the [Installing Joomla via the CLI](https://docs.joomla.org/J4.x:CLI_Update#Installing_Joomla.21_via_the_CLI) documentation

Because you use Docker image please follow the information from the image description. https://hub.docker.com/_/joomla
If I right understood it will "pre install" joomla for you.

For bugs in the image please reaport https://github.com/joomla-docker/docker-joomla/

**Comment by alikon** [comment]
@Fedik  `php cli/joomla.php core:install` has never been landed in core,  that doc  page must be amended
starting for 4.3 you can install from cli with `php installation/joomla.php install`

so imho it's not a bug more a doc issue

**Comment by Fedik** [comment]
Okay :)

**Comment by nickyeoman** [comment]
Yes, I think this is a documentation issue.  The page provided by @alikon [J4.xJoomla CLI Installation](https://docs.joomla.org/J4.x:Joomla_CLI_Installation) is what I was trying to do.

Tested the above documentation page with dockerhub image joomla:4.3.1 and the following bash script, works perfect.
```
#!/bin/bash
# Load .env
set -o allexport
source .env
set +o allexport

DB_PREFIX=$(LC_CTYPE=C tr -dc '[:alpha:]' < /dev/urandom | head -c 3)_

docker-compose exec -u www-data joomla php installation/joomla.php install \
  --site-name=$PROJECTNAME \
  --admin-email=$EMAIL \
  --admin-user=$JOOMLA_USER \
  --admin-username=$JOOMLA_USER \
  --admin-password=$JOOMLA_PASSWORD \
  --db-type=mysqli \
  --db-host=mariadb \
  --db-user=$MYSQL_USER \
  --db-pass=$MYSQL_PASSWORD \
  --db-name=$MYSQL_DATABASE \
  --db-prefix=$DB_PREFIX \
  --db-encryption=0
```

### Docker note

Just to clarify ```If I right understood it will "pre install" joomla for you.``` -@Fedik

That statement is not correct, the [DockerFile](https://github.com/joomla-docker/docker-joomla/blob/master/Dockerfile.template) only extracts the zip file and gives you a ready installation (unless the volume is already full then it skips installation).
```
RUN set -ex; \
	curl -o joomla.tar.bz2 -SL {{ env.joomlaPackage }}; \
	echo "$JOOMLA_SHA512 *joomla.tar.bz2" | sha512sum -c -; \
	mkdir /usr/src/joomla; \
	tar -xf joomla.tar.bz2 -C /usr/src/joomla; \
	rm joomla.tar.bz2; \
	chown -R www-data:www-data /usr/src/joomla
```

I'm not even sure if it does anything with the ENV it asks for on the dockerhub page.  But as you say that's an issue for: https://github.com/joomla-docker/docker-joomla/

My above code could be run from a custom dockerfile instead of docker-compose

# Conclusion

Yes, this is a documentation issue with the page: https://docs.joomla.org/J4.x:CLI_Update#Installing_Joomla.21_via_the_CLI 

As for a solution, I'm not sure if it should be removed, as I like the idea of a json file.

**Comment by rdeutz** [comment]
not an core issue
